packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -165,7 +170,7 @@ export class ApolloServerBase<
       mockEntireSchema,
       experimental_approximateDocumentStoreMiB,
       ...requestOptions
-    } = config;
+    } = this.config;
```

Comment:
`requestOptions` will get `nodeEnv` here. This means that it will _always_ be available when this later gets used for the `GraphQLServerOptions`.